docs/testing_servers.md

```diff
@@ -0,0 +1,340 @@
+  * `message_receive_limit` specifies the maximum size in bytes for a message. If this value is non-zero, servers should reject
+     any message from a client that is larger than the size indicated.
+
+* Once the server is started, your implementation should then write a `ServerCompatResponse` message to `stdout`. This will
+  provide the conformance runner with details about your running server.
+
+  * `host` should be set with the host where your server is running. 
+  * `port` should be set with the port number where your server is listening.
+  * `pem_cert` should contain the self-signed certificate generated by your server if `use_tls` was set to `true`. Clients
+     will verify this certificate when connecting via TLS. If `use_tls` was set to `false`, this should always be empty.
+
+## Implementing the ConformanceService
+
+The ConformanceService defines a series of endpoints that are meant to exercise all types of RPCs. The details
+in the requests will specify various attributes that the server should handle, such as determining response headers and 
+trailers to return, any errors to throw, and any data to respond with. All request types contain a response definition 
+which is used to instruct the server how to respond to the request. This response definition can also be unset entirely
+and servers should respond in a fashion specific to their RPC type. The various approaches are outlined below.
+
+Servers must also echo back the received request data in their responses and the method for doing so will vary depending 
+on the type of RPC. The `ConformancePayload` message is a field on all RPC response types. Servers will use this field 
+to echo back these details. This will include:
+
+* Any response data
+* Information observed from the request such as:
+  * Request headers
+  * Any timeout specified
+  * All request bodies received
+  * Any query parameters observed (only applicable for GET operations such as `IdempotentUnary`)
+
+In all, there are six total endpoints you will need to implement. Below is a brief description of each with helpful pseudocode
+for your implementation.
+
+### Unary
+
+The `Unary` endpoint is a unary operation that accepts a single request of type `UnaryRequest` and returns a single 
+response of type `UnaryResponse`. 
+
+The `UnaryRequest` contains a `response_definition` that tells the server how to respond. This definition contains a 
+`oneof` which specifies whether the server should return valid response data or return an error. Servers should also 
+allow this response definition to be unset. In which case, they should set no response headers or trailers and return 
+no response data. The returned conformance payload should only contain the observed request information.
+
+**Pseudocode**
+
+```text
+read the request
+
+capture any request headers and the actual request body as request info
+
+if a response definition was specified then
+  if response definition specifies valid response data then
```

Comment:
Maybe we can tweak this condition and then swap the two branches? I think it would be more clear if it were more like "if response definition specifies an error then"